src/Browser/Json.php

```diff
@@ -33,6 +48,63 @@ public function assertMatches(string $expression, $expected): self
         return $this;
     }
 
+    /**
+     * @param string $selector JMESPath selector
+     */
+    public function assertHas(string $selector): self
+    {
+        Assert::try(fn() => $this->search("length({$selector})"));
+
+        Assert::pass();
+
+        return $this;
+    }
+
+    /**
+     * @param string $selector JMESPath selector
+     */
+    public function assertMissing(string $selector): self
+    {
+        try {
+            $this->search("length({$selector})");
+        } catch (\RuntimeException $e) {
+            Assert::pass();
+
+            return $this;
+        }
+
+        Assert::fail("Selector \"{$selector}\" does exists.");
+    }
+
+    /**
+     * @phpstan-param callable(Json $json): void $assert
```

Comment:
not sure about the `: void` since arrow function always return something...

Reply:
when called like this:
```php
(new Json(''))->assertThat('x', fn(Json $json) => $json->is(''));
```

Reply:
of course, you're right
but phpstan does not like this :shrug:

Reply:
You're right, let's leave as is.

Reply:
I've tests with an arrow function and with an old-fashioned anonymous function, phpstan is OK with both with `:mixed`